android/sdl_android/src/main/java/com/smartdevicelink/managers/video/VideoStreamManager.java

```diff
@@ -648,6 +773,101 @@ void createTouchScalar(ImageResolution resolution, DisplayMetrics displayMetrics
+        validCapabilities.add(preferredCapability);
+        for (VideoStreamingRange range: ranges) {
+
+            Integer constraintHeightMax = range.getMaxResolution().getResolutionHeight();
+            Integer constraintHeightMin = range.getMinResolution().getResolutionHeight();
+
+            for (VideoStreamingCapability capability : originalAdditionalCapabilities) {
+                double diagonal;
+                if (capability.getPreferredResolution() == null
+                        || capability.getPreferredResolution().getResolutionHeight() == null
+                        || capability.getPreferredResolution().getResolutionWidth() == null) {
+                    continue;
+                }
+                if (capability.getDiagonalScreenSize() == null) {
+                    diagonal = parameters.getPreferredDiagonal();
+                } else {
+                    diagonal = capability.getDiagonalScreenSize();
+                }
+
+                if (range.getMinScreenDiagonal() > diagonal) {
+                    continue;
+                }
+
+                if (!isAspectRatioInRange(range.getAspectRatio(), capability.getPreferredResolution())) {
+                    if (constraintHeightMax == null && constraintHeightMin == null) {
+                        continue;
+                    }
+                }
+
+                if (!isImageResolutionInRange(range.getMinResolution(), range.getMaxResolution(), capability.getPreferredResolution())) {
+                    continue;
+                }
+
+                validCapabilities.add(capability);
+            }
+        }
+
+        return validCapabilities;
+    }
+
+    public Boolean isImageResolutionInRange(Resolution minResolution, Resolution maxResolution, ImageResolution currentResolution) {
+
+        Integer constraintHeightMax = maxResolution.getResolutionHeight();
+        Integer constraintHeightMin = minResolution.getResolutionHeight();
+        Integer constraintWidthMax = maxResolution.getResolutionWidth();
+        Integer constraintWidthMin = minResolution.getResolutionWidth();
+        Integer resolutionHeight = currentResolution.getResolutionHeight();
+        Integer resolutionWidth = currentResolution.getResolutionWidth();
+        if (currentResolution.getResolutionHeight() > 0 && currentResolution.getResolutionWidth() > 0 && constraintHeightMax != null && constraintHeightMin != null) {
+            if (!(resolutionHeight >= constraintHeightMin && resolutionHeight <= constraintHeightMax)) {
+                return false;
+            }
+
+            if (!(resolutionWidth >= constraintWidthMin && resolutionWidth <= constraintWidthMax)) {
+                return false;
+            }
+        }
+
+        // TODO check what if dev provided invalid constraints
+        return true;
+    }
+
+    public Boolean isAspectRatioInRange(AspectRatio aspectRatio, ImageResolution currentResolution) {
```

Comment:
@kostyaBoss Per the proposal, `isAspectRatioInRange` should be defined as `public Boolean isAspectRatioInRange(VideoStreamingRange range, ImageResolution currentResolution)`.

Your definition is different because you take in an `AspectRatio` object instead of a `VideoStreamingRange` object. Do you need to change yours to match up with the proposal?

Reply:
@santanamk in the proposal it was just a section where a preview example was provided. To keep libraries mirroring each other we've agreed to make this method accept two arguments